### PR TITLE
added ability to key in flexible dates. tests added and regression testing complete.

### PR DESCRIPTION
**Pull request by jothipillay**

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/FlexibleDateParser.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+                || (keyword.equals(CliSyntax.PREFIX_PAST.toString()))) {
+                return parseKeyword(keyword, arguments.trim());
+            } else {
+                throw new ParseException(Deadline.MESSAGE_CONSTRAINTS);
+            }
+        }
+    }
+
+    /**
+     * @param keyword they keyword which indicates if it is current (this), past(last) or future(next).
```
Comment (jeffgan96) [inline src/main/java/seedu/address/logic/parser/FlexibleDateParser.java:78]: spelling haha

File: src/main/java/seedu/address/logic/parser/FlexibleDateParser.java [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+
+        final Matcher matcher = BASIC_FLEXIDATE_FORMAT.matcher(secondPart.trim());
+        if (!matcher.matches()) {
+            throw new ParseException(Deadline.MESSAGE_CONSTRAINTS);
+        }
+
+        final String weekOrMonth = matcher.group("keyword").toLowerCase();
+        final String arguments = matcher.group("arguments");
+
+        if (weekOrMonth.equals(CliSyntax.PREFIX_WEEK.toString())) {
```
Comment (jeffgan96) [inline src/main/java/seedu/address/logic/parser/FlexibleDateParser.java]: can condense

File: src/main/java/seedu/address/logic/parser/FlexibleDateParser.java (hunk 1 above)
Comment (jothipillay) [reply src/main/java/seedu/address/logic/parser/FlexibleDateParser.java:78]: thanks for the keen eye

Comment (jothipillay) [reply src/main/java/seedu/address/logic/parser/FlexibleDateParser.java:78]: @jeffgan96

File: src/main/java/seedu/address/model/util/CalendarDatesInMonth.java [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    private static final int AUGUST = 8;
+    private static final int AUG_MAX_DAYS = 31;
+    private static final int SEPTEMBER = 9;
+    private static final int SEP_MAX_DAYS = 30;
+    private static final int OCTOBER = 10;
+    private static final int OCT_MAX_DAYS = 31;
+    private static final int NOVEMBER = 11;
+    private static final int NOV_MAX_DAYS = 30;
+    private static final int DECEMBER = 12;
+    private static final int DEC_MAX_DAYS = 31;
```
Comment (jeffgan96) [inline src/main/java/seedu/address/model/util/CalendarDatesInMonth.java]: condense into etc JAN_MAX_DAYS, MARCH_MAX_DAYS = 31

**Pull request merged**